rskj-core/src/main/java/co/rsk/peg/bitcoin/UtxoUtils.java

```diff
@@ -0,0 +1,83 @@
+package co.rsk.peg.bitcoin;
+
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.bitcoinj.core.VarInt;
+import com.google.common.primitives.Bytes;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.spongycastle.util.encoders.Hex;
+
+public class UtxoUtils {
+
+    private UtxoUtils() {
+    }
+
+    /**
+     * Decode a {@code byte[]} of encoded outpoint values.
+     *
+     * @param encodedOutpointValues
+     * @return {@code List<Coin>} the list of outpoint values decoded in the same ordered they were
+     * before encoding. Or an {@code Collections.EMPTY_LIST} when {@code encodedOutpointValues} is
+     * {@code null} or {@code empty byte[]}.
+     */
+    public static List<Coin> decodeOutpointValues(byte[] encodedOutpointValues) {
+        if (encodedOutpointValues == null || encodedOutpointValues.length == 0) {
+            return Collections.emptyList();
+        }
+        int offset = 0;
+        List<Coin> outpointValues = new ArrayList<>();
+
+        while (encodedOutpointValues.length > offset) {
+            VarInt valueAsVarInt;
+            try {
+                valueAsVarInt = new VarInt(encodedOutpointValues, offset);
+            } catch (Exception ex) {
+                throw new InvalidOutpointValueException(
+                    String.format("Invalid value with invalid VarInt format: %s",
+                        Hex.toHexString(encodedOutpointValues).toUpperCase()
```

Comment:
Why upper case?

Reply:
Because [VarInt documentation](https://wiki.bitcoinsv.io/index.php/VarInt) is expressed using upper case.